server/network.go

```diff
@@ -171,22 +186,51 @@ func newLocalNetwork(opts localNetworkOptions) (*localNetwork, error) {
 	}, nil
 }
 
-func createConfigFileString(nodeConfig string, logLevel string, logDir string, dbDir string, pluginDir string, whitelistedSubnets string) (string, error) {
-	var jsonContent map[string]interface{}
-	if err := json.Unmarshal([]byte(nodeConfig), &jsonContent); err != nil {
-		return "", err
+// mergeNodeConfig evaluates the final node config.
+// defaultConfig: map of base config to be applied
+// globalConfig: map of global config provided to be applied to all nodes. Overrides defaultConfig
+// customConfig: a custom config provided to be applied to this node. Overrides globalConfig and defaultConfig
+// returns final map of node config entries
+func mergeNodeConfig(baseConfig map[string]interface{}, globalConfig map[string]interface{}, customConfig string) (map[string]interface{}, error) {
+	var jsonCustom map[string]interface{}
+
+	// merge, overwriting entries in default with the global ones
+	for k, v := range globalConfig {
+		baseConfig[k] = v
 	}
+
+	if customConfig != "" {
```

Comment:
Ok now I see that custom node configs len is not invalidating number of nodes param... That is correct behaviour for me. Shall only change then the description of the cobra command.

Reply:
https://github.com/ava-labs/avalanche-network-runner/pull/134/files/f7eb4b273a525fbdb1333d0a76b33e7f8666ccb5#diff-bcef29333b5dd1d156e40416d98c682d7b3d16d1d7485ca8ad885ba25958d39f